components/ConvertSteps/ManageStep.js

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import Step from './Step'
+
+function ManageStep({
+  title,
+  number,
+  handleTx,
+  active,
+  onSuccess,
+  onError,
+  onHashCreation,
+  retry,
```

Comment:
Perhaps we can rename this to something else. In particular, I'm finding it difficult to understand the differences between the `retry` prop and the `retrying` state (perhaps this is something `getDerivedStateFromProps()` would help with?)

Reply:
I agree, this isn't clear. Like my previous comment I think this problem could be alleviated by a small refactor in how the state is managed.